room/compiler/src/main/kotlin/androidx/room/processor/DatabaseProcessor.kt

```diff
@@ -107,7 +107,11 @@ class DatabaseProcessor(baseContext: Context, val element: XTypeElement) {
                 views = views,
                 daoMethods = daoMethods,
                 exportSchema = dbAnnotation.value.exportSchema,
-                enableForeignKeys = hasForeignKeys)
+                enableForeignKeys = hasForeignKeys,
+                typeConverterFactories = HashSet(context.typeConverterFactories))
+
+        context.typeConverterFactories.clear()
```

Comment:
This mutable map is still a bit scary, what we can do is move the set to a property in the `DatabaseProcessor` and then let the `Context` have a function that `CustomComnverterProcessor` can call into that is provided by `DatabaseProcessor` that adds to the set.

In other words:
```
class Context (
  ...
  val addTypeConverterFactory: (XType) -> Unit
) {
  ...
  fun fork(..., addTypeConverterFactory: (XType) -> Unit = {  }) {
    ...
  }
}
```
then:
```
class DatabaseProcessor(...) {
    val context = baseContext.fork(
        element = element,
        addTypeConverterFactory = { typeConverterFactories.add(it) }
    )
    private val typeConverterFactories = mutableSetOf<XType>()
}
```

Reply:
Yeah it's much cleaner! Thanks 👍

Reply:
There is a slight problem though - it doesn't work 😅 
I think the problem is that `fork` calls `CustomConverterProcessor.findConverters(this, element)` and after that it creates a sub context. So `findConverters` is called before `addTypeConverterFactory` is assigned. I may be wrong though.
Trying to make it work now.

Reply:
Maybe it could be done by passing `addTypeConverterFactory: (XType) -> Unit` as a third parameter to the `findConverters` function. Just like `element` param.

Reply:
Passing `addTypeConverterFactory` to `CustomConverterProcessor.findConverters` sounds totally reasonable, then we should make the property `private` in `Context` to avoid misuse.

Reply:
Yup! 👍